### PR TITLE
feat(kre-go): Add key value stores

### PR DESCRIPTION
**Pull request by kafkaphoenix**
Implementation of key value storage for each scope (Node, Workflow, Project)

## Description
Add method Set, Get and Delete for key value storage. Set and Delete accept an optional scope if none is given it will use the default scope (Node), get will try to get the value from one scope is none is given starting by Node and going upwards

## Motivation and Context
A key value storage will let us to store configurations and share them.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other changes (ci configuration, documentation or any other kind of changes)

## Checklist:

- [ ] I have created tests for my code changes, and the tests are passing.
- [ ] I have executed the pre-commit hooks locally.
- [ x] My change requires a change to the documentation (create a new issue if the documentation has not been updated).
- [ ] I have updated the documentation accordingly.

Co-authored-by: @alexrodfe 
Co-authored-by: @JYisus
Co-authored-by: @dfinteligenz

### CODE REVIEW

File: kre-go/runner.go [hunk 1]
```diff
@@ -212,14 +218,69 @@ func (r *Runner) deleteObject(key string) error {
...
+	}
+
+	return nil
+}
+
+func (r *Runner) getConfig(key string, scopeOpt []Scope) (string, error) {
+	if len(scopeOpt) > 0 {
+		return r.getConfigFromScope(key, scopeOpt[0])
+	} else {
+		allScopes := []Scope{ScopeNode, ScopeWorkflow, ScopeProject}
```
Comment (JYisus) [inline kre-go/runner.go:247]:
```suggestion
		allScopes := [3]Scope{ScopeNode, ScopeWorkflow, ScopeProject}
```

File: kre-go/runner.go [hunk 2]
```diff
@@ -212,14 +218,69 @@ func (r *Runner) deleteObject(key string) error {
...
+			}
+		}
+
+		return "", fmt.Errorf("error retrieving config with key %s, not found in any key-value store", key)
+	}
+}
+
+func (r *Runner) getConfigFromScope(key string, scope Scope) (string, error) {
+	value, err := r.kvStoresMap[scope].Get(key)
+
```
Comment (JYisus) [inline kre-go/runner.go:262]: `if err != nil` clauses should be in the next line of the function that returned the error

File: kre-go/runner.go [hunk 3]
```diff
@@ -212,14 +218,69 @@ func (r *Runner) deleteObject(key string) error {
...
+func (r *Runner) deleteConfig(key string, scope Scope) error {
+	kvStore, ok := r.kvStoresMap[scope]
+	if !ok {
+		return fmt.Errorf("could not find key value store given scope %s", scope)
+	}
+
+	err := kvStore.Delete(key)
+	if err != nil {
+		return fmt.Errorf("error deleting value with key %s from the key-value store: %w", key, err)
```
Comment (JYisus) [inline kre-go/runner.go:278]:
```suggestion
		return fmt.Errorf("error deleting value with key %q from the key-value store: %w", key, err)
```

**Review by JYisus — CHANGES_REQUESTED** [review]

**Comment by alexrodfe** [comment]
Moved on to new branch so merge conflicts with object store would be easier to handle with. New PR regarding key value store is here:
https://github.com/konstellation-io/kre-runners/pull/174

**Pull request closed without merging**